frontend/public/components/monitoring/alert-manager-utils.tsx

```diff
@@ -4,28 +4,45 @@ import { Base64 } from 'js-base64';
 import { safeLoad, safeDump } from 'js-yaml';
 
 import { k8sPatch, K8sResourceKind } from '../../module/k8s';
-import {
-  AlertManagerConfig,
-  AlertManagerReceiver,
-  AlertManagerRoute,
-} from './alert-manager-config';
+import { AlertManagerConfig } from './alert-manager-config';
 import { SecretModel } from '../../models';
 
-export const DEFAULT_RECEIVER_LABEL = 'All (default receiver)';
+export const receiverTypes = {
+  pagerduty_configs: 'PagerDuty',
+  webhook_configs: 'Webhook',
+  email_configs: 'Email',
+  slack_configs: 'Slack',
+};
 
-export const getAlertManagerConfig = (secret: K8sResourceKind, setErrorMsg) => {
+const decodeAlertManagerYaml = (secret: K8sResourceKind, setErrorMsg) => {
   const alertManagerYaml = _.get(secret, ['data', 'alertmanager.yaml']);
-  let config: AlertManagerConfig;
+  let yaml: string = '';
 
   if (_.isEmpty(alertManagerYaml)) {
     setErrorMsg(
       'Error: alertmanager.yaml not found in Secret "alertmanager-main", in namespace "openshift-monitoring"',
     );
-    return config;
+    return yaml;
   }
 
   try {
-    config = safeLoad(Base64.decode(alertManagerYaml));
+    yaml = Base64.decode(alertManagerYaml);
+  } catch (e) {
+    setErrorMsg(`Error decoding alertmanager.yaml: ${e}`);
+  }
+
+  return yaml;
+};
+
+export const getAlertManagerYAML = (secret: K8sResourceKind, setErrorMsg): string => {
```

Comment:
`getAlertManagerYAML` is an alias for `decodeAlertManagerYaml`, so maybe just export `decodeAlertManagerYaml` instead?

Reply:
I like the symmetry of  `getAlertManagerYAML` and `getAlertManagerConfig`.  One gives yaml string, the other gives config obj.

Reply:
OK. You could keep the name `getAlertManagerYAML`, but still merge the two functions. I think this function actually just being an alias for another is a bit confusing.

Reply:
fixed